### DISCUSSION TITLE
Grid utilities does not working on NextJS

### DISCUSSION THREAD

**Discussion by hmtri1011** [tailwindlabs/tailwindcss#5551] [Help] [answered]
**What version of Tailwind CSS are you using?**
v2.2.15

**What build tool (or framework if it abstracts the build tool) are you using?**
Postcss 8.3.6, Next.js 11.1.0

**What version of Node.js are you using?**
v15.9.0

**What browser are you using?**
Chrome

**What operating system are you using?**
macOS

**Describe your issue**
I'm using nextJS v11.1.0 and tailwind CSS v2.2.15, and all utilities are working except grid (grid, gridTemplateColumns, ...). This is super super weird issue.

![image](https://user-images.githubusercontent.com/16288424/134024927-8bdcb278-389f-46e0-9df3-685b34f468b6.png)

**Comment by adamwathan** [comment, marked as answer]
Hey! Please provide an example project — we use Next.js for tailwindcss.com for example and use the grid utilities everywhere so it must be a project-specific configuration issue.

**Reply by hmtri1011** [reply]
I follow the nextJS installation guide (https://tailwindcss.com/docs/guides/nextjs) and am able to use all utilities, except the grid, className="grid grid-cols-3" has no effect for me. I don't have any custom CSS in global.css or tailwind.config.js :(

**Reply by hmtri1011** [reply]
![image](https://user-images.githubusercontent.com/16288424/134027330-f0bb3d27-d19b-4fc7-86fb-d70337963f99.png)
margin works

**Reply by hmtri1011** [reply]
![image](https://user-images.githubusercontent.com/16288424/134027450-6107ade1-7539-4068-bee7-6f2a0cb334e4.png)
but not for grid and grid-cols-3, no styles definition

**Reply by simonswiss** [reply]
Hey! Please share a real code example where you're able to reproduce this if you want help – it's impossible to troubleshoot from screenshots, and as Adam mentioned, we use grid utilities on Next.js projects without problems.

I see classes like `bg-neutral-light`, `bg-blue-dark` and `rounded-banner` so I assume you have some customisations in your Taliwind Config, and perhaps something went wrong there?

**Reply by hmtri1011** [reply]
This is my company project so I cannot share the code. I have used tailwind for 2 years so I am pretty sure that my configuration and customization do not have any problems.

I figure out the problem, enable Just In Time mode will make the grid utilities not working on NextJS. By disabling JIT mode, it will be working fine.

**Reply by hmtri1011** [reply]
![image](https://user-images.githubusercontent.com/16288424/134139116-e2185d26-6dd9-45b1-a2d0-dda0c8db6baa.png)
This is my tailwind config, disable JIT mode will make the grid utilities working again, I think this is a very weird problem

**Reply by simonswiss** [reply]
Hey!

I understand you cannot share your company project's code, but perhaps you can recreate a minimal reproduction of the same scenario (Next.js, Tailwind, same Tailwind config, PostCSS config, CSS input etc), and see if the problem persists?

I have used CSS grid utilities with Next.js and Tailwind `JIT` numerous times without a problem, so there is definitely a weird thing going here, and it's impossible to figure out what it is without some code to run and tinker with 👍

**Reply by hmtri1011** [reply]
thank guys, it is really weird, can close now

**Reply by PiotrNap** [reply]
@hmtri1011 I had the same problem. Try deleting your className attribute and pasting it again. This has worked for me.

**Reply by hmtri1011** [reply]
@PiotrNap I think JIT does not stable for NextJS. There has a lot of weird issues.

for example: I have Home component in containers folder and in `pages/home.tsx` if I do
`export { default } from 'containers/Home'`
then there will have a problem where some of className does not affected (mt, mb, md:mt)

But if in `pages/home.tsx` I copy exactly all the code from `containers/Home`, then It works perfectly, lol. Dunno what is going wrong here.

**Reply by hmtri1011** [reply]
and that is the way I solved my problem

**Reply by PiotrNap** [reply]
Glad to hear you solved it, way to go brother.

**Reply by firhanram** [reply]
I had same issue like you. I created a new folder for my layouts. Then, tailwind css didnt work in this folder. My solution was I added my layouts folder path to `tailwnd.config.js` on `content: []` and it works!. Hope, this can help.

### Before
![before](https://user-images.githubusercontent.com/59257112/146949039-bfe984d2-23aa-42d6-97e6-baf082ab7722.png)

### After
![tailwindcss after](https://user-images.githubusercontent.com/59257112/146949069-c9208e0c-936a-46fe-bdd5-e735ef4370d6.png)

**Reply by SciFiTy10** [reply]
thank you @firhanram! This was the exact problem with my setup as well. I had to add the layouts path to `tailwind.config.js`

**Reply by filip-dahlberg** [reply]
I'm also experiencing the very weird problem using Tailwind with Nextjs. I can only get grid-cols-4 to work, grid-cols-2 does not appear in the browser's css style.
<img width="654" alt="Screenshot 2022-02-12 at 09 05 19" src="https://user-images.githubusercontent.com/37364201/153702949-0a1f08c8-2f15-4fc9-a693-c283aad2c54c.png">
Figure 1: grid-cols-2 not working, see style missing grid-template-columns.

![Uploding Screenshot 2022-02-12 at 09.06.15.png…]()
Figure 2: grid-cols-4 is correctly working, see style including grid-template-columns. 


WORKAROUND:
Add the folder path to 'content' located in folder 'pages', which in my case is 'projects'. This seems to prove a bug since the pages/** should be sufficient AFAIK.
<img width="603" alt="Screenshot 2022-02-12 at 09 22 32" src="https://user-images.githubusercontent.com/37364201/153703512-b113fe50-af6a-4ee4-9a26-6cfdc1d6f983.png">

**Reply by sinamics** [reply]
had the exact same issue, the workarround mentioned above is working for me as well.

**Reply by dangtony98** [reply]
Had a similar issue; turns out I had a typo — just had to change ./components**/*.{js,ts,jsx,tsx} to ./components/**/*.{js,ts,jsx,tsx} and it worked just fine!

**Reply by Ntwali-Josue** [reply]
Thank you @firhanram, this solved my issue too

**Reply by naofunky** [reply]
Thank you @firhanram , this helped my issue too.
In my case, needed to change relative path lol

**Reply by abubakar471** [reply]
> me mode will make the grid utilities not

that worked for me
